### PR TITLE
Feature/operation white list

### PR DESCRIPTION
**Pull request by DoraALin**
file baed whitelist. Extra config "access_control_file" introduced to specify path to access control white list, which is a yaml of following format:

<pre>
#enable white list, false means disable
enable: true
#administrators map, with user name as key, and enable/disable as val.
admin:
 tom: false
 jack: false
 lee: true
 lucky: false

</pre>

### CODE REVIEW

File: nsqadmin/http.go [hunk 1]
```diff
@@ -158,6 +158,23 @@ func parseAccessToken(r *http.Request) (token string, ok bool) {
...
+func (s *httpServer) adminCheck(f http_api.APIHandler) http_api.APIHandler {
+	return func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) (interface{}, error) {
+		//check user info
+		if s.ctx.nsqadmin.IsAuthEnabled() {
+			u, err := s.getUserInfo(w, req)
+			if err != nil {
+				s.ctx.nsqadmin.logf("error in fetching user model %v", err)
+				return nil, http_api.Err{http.StatusInternalServerError, "fail to find associated user info"}
+			}
+			if !(u.IsLogin() && u.IsAdmin()) && !s.validAccessToken(req) {
```
Comment (absolute8511) [inline nsqadmin/http.go]: IsAdmin should be treated as login.

File: nsqadmin/auth.go [hunk 2]
```diff
@@ -118,7 +116,11 @@ func (u *CasUserModel) GetUserRole() string {
 }
 
 func (u *CasUserModel) IsAdmin() bool {
-	return u.Admin
+	if ac := u.ctx.nsqadmin.ac; ac == nil {
+		return true
+	} else {
+		return ac.IsAdmin(u.UserName)
```
Comment (absolute8511) [inline nsqadmin/auth.go]: IsAdmin should check whether we are login.

File: nsqadmin/whitelist.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+					//update access control map
+					ac.tryUpdateControlFile()
+				}
+			case <-ac.tStopChan:
+				{
+					goto stop
+				}
+			}
+		}
+stop:
```
Comment (absolute8511) [inline nsqadmin/whitelist.go]: Maybe should use defer instead goto

File: nsqadmin/whitelist.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+		ac.wg.Done()
+	}()
+}
+
+func (ac *YamlAccessControl) Stop() {
+	ac.updateTicker.Stop()
+	ac.tStopChan <- 0
```
Comment (absolute8511) [inline nsqadmin/whitelist.go]: should close chan here to notify all.

**Comment by absolute8511** [comment]
CI missing dependency for `github.com/astaxie/beego/config/yaml`

**Comment by DoraALin** [comment]
@absolute8511 for your review.

**Comment by absolute8511** [comment]
fixed #33

**Pull request merged**